pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1961,6 +2010,44 @@
           %leave  ~
       ==
     --
+  ::  +add-cache: remember a cache mapping
+  ::
+  ++  add-cache
+    |=  [=path =cache-entry]
+    ^-  [(list move) server-state]
+    =/  cache-entries  (fall (~(get of cache.state) path) ~)
+    =.  cache.state  (~(put of cache.state) path [[~ cache-entry] cache-entries])
+    `state
+  ::  +del-cache: forget a cache mapping
+  ::
+  ++  del-cache
+    |=  =path
+    ^-  [(list move) server-state]
+    =/  cache-entries  (fall (~(get of cache.state) path) ~)
+    ::  replace most recent entry with ~
+    ::
+    =?  cache.state  ?=(^ cache-entries)
+      (~(put of cache.state) path [~ t.cache-entries])
+    `state
```

Comment:
Should these also `%give %cache-changed` (maybe depending on the presence of a relevant binding)?

Reply:
Yes.

Reply:
Should a gift be given even when deleting an entry whose case is not the most recent?

Maybe it's just me who's confused, but the intended behavior/use case of `%cache-changed` is not super clear to me, particularly in relation to the cache/cache-binding distinction.

Reply:
Imo the semantics of the `%cache-changed` gift are that when anything in the `cache-binding` data structure changes, a gift should be sent. Note that if an item in the `cache` changes that is currently referenced by `cache-binding`, that counts as changing `cache-binding`.

Reply:
ok, added